src/store/sagas/visualization-saga.test.js

```diff
@@ -289,6 +289,56 @@ describe('visualization-saga', () => {
       })
     })
   })
+
+  describe('deleteVisualization', () => {
+    const initialState = {}
+
+    describe('successfully', () => {
+      var dispatched = []
+      beforeEach(async () => {
+        AuthenticatedHTTPClient.delete = jest.fn(() => ({ status: 204, data: {} }))
+        dispatched = await recordSaga(deleteVisualization, visualizationDelete({ id: "3" }), initialState)
+      })
+
+      it('signals the visualization is deleted', () => {
+        expect(dispatched).toContainEqual(visualizationDeleteSuccess())
+      })
+
+      it('dispatches a load user visualization list action', () => {
+        expect(dispatched).toContainEqual(visualizationsLoadAll())
+      })
+    })
+
+    describe('with a non-success status code', () => {
+      const nonSuccessStatusCode = 400
+      var dispatched = []
+      beforeEach(async () => {
+        AuthenticatedHTTPClient.delete = jest.fn(() => ({ status: nonSuccessStatusCode }))
+        dispatched = await recordSaga(deleteVisualization, visualizationDelete({ id: "2" }), initialState)
+      })
+
+      it('signals the visualization deletion failed', () => {
+        expect(dispatched).toContainEqual(visualizationDeleteFailure(nonSuccessStatusCode))
+      })
+
+      it('dispatches a load user visualization list action', () => {
+        expect(dispatched).toContainEqual(visualizationsLoadAll())
+      })
+    })
+
+    describe('with a thrown error', () => {
+      const errorMessage = 'WRONG'
+      var dispatched = []
+      beforeEach(async () => {
+        AuthenticatedHTTPClient.delete = jest.fn(() => { throw new Error(errorMessage) })
+        dispatched = await recordSaga(deleteVisualization, visualizationDelete({ id: "1" }), initialState)
+      })
+
+      it('signals the visualization is unavailable', () => {
+        expect(dispatched).toContainEqual(visualizationDeleteFailure(errorMessage))
+      })
```

Comment:
Do we also want to confirm the loadAll here?

Reply:
Yes, I'll add that